### PR TITLE
Merge snapshot charts

### PR DESCRIPTION
**Pull request by phoenix-bjoern**
Merges snapshot-validation-webhook into snapshot-controller chart (implements #35).

### CODE REVIEW

File: charts/snapshot-controller/Chart.yaml [hunk 1]
```diff
@@ -10,9 +10,12 @@ maintainers:
 description: |
   Deploys a Snapshot Controller in a cluster. Snapshot Controllers are often bundled with the Kubernetes distribution,
   this chart is meant for cases where it is not.
+  Also deploys the Snapshot Validation Webhook and configures your cluster to validate every `VolumeSnapshot` and
+  `VolumeSnapshotContent` resource by sending it to the webhook.
 keywords:
   - storage
   - snapshot
+  - validation
```
Comment (WanzenBug) [inline charts/snapshot-controller/Chart.yaml:18]: There is a `annotations` section at the end of this file which needs to be updated: we no longer need to recommend the webhook chart.

File: charts/snapshot-validation-webhook/README.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # snapshot-validation-webhook
 
+> **Discontinuing notice**: The chart has been merged with the [*snapshot-controller*](../snapshot-controller/README.md) chart
+> in the same repository. This chart has been archived.
+
 Deploys the [snapshot-validation-webhook](https://github.com/kubernetes-csi/external-snapshotter/#validating-webhook)
```
Comment (WanzenBug) [inline charts/snapshot-validation-webhook/README.md:6]: Helm also seems to support a `deprecated:` keyword in Chart.yaml. We should set that and bump the version to trigger a new release. Presumably users would then get a warning about the deprecation.

File: charts/snapshot-controller/README.md [hunk 3]
```diff
@@ -63,26 +64,28 @@ The upgrade procedure can be summarized by the following steps:
...
    ```
 
 ## Configuration
 
+### Snapshot controller
 The following options are available:
 
 | Option                      | Usage                                                                                                                  | Default                                                                                            |
 |-----------------------------|------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------|
+| `enabled`                   | Toggle to disable the deployment of the snapshot controller.                                                           | `true`                                                                                             |
```
Comment (WanzenBug) [inline charts/snapshot-controller/README.md:88]: Should probably make it clear somewhere that these values need to be prefixed with `controller` (and `webhook` for the options below)

File: charts/snapshot-controller/templates/_helpers.tpl [hunk 4]
```diff
@@ -49,9 +49,78 @@ app.kubernetes.io/instance: {{ .Release.Name }}
...
+{{- end }}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "snapshot-validation-webhook.fullname" -}}
+{{- if contains .Chart.Name .Release.Name }}
+{{- .Release.Name | trunc 63 | trimSuffix "-" }}
```
Comment (WanzenBug) [inline charts/snapshot-controller/templates/_helpers.tpl:73]: When I try to install the chart, I get 
```
Error: INSTALLATION FAILED: serviceaccounts "snapshot-controller" already exists
```

I believe this default needs to be changed to include a `-webhook` suffix or similar.

**Review by WanzenBug — COMMENTED** [review]
Thank you so much for this! :raised_hands: 

Just needs some work, because currently it does not want to deploy for me.

File: charts/snapshot-controller/templates/_helpers.tpl (hunk 4 above)
Comment (phoenix-bjoern) [reply charts/snapshot-controller/templates/_helpers.tpl:73]: @WanzenBug If the deployment name is "snapshot-controller" then everything becomes "snapshot-controller" without any suffix/prefix for the *snapshot-validation-webhook* components.
Will pin it to `snapshot-validation-webhook` in that case.

**Review by WanzenBug — APPROVED** [review]
LGTM

**Pull request merged**